prov/ansible/execution.go

```diff
@@ -771,9 +773,11 @@ func (e *executionCommon) generateHostConnection(ctx context.Context, buffer *by
 		sshCredentials := e.getSSHCredentials(ctx, host, true)
 		buffer.WriteString(fmt.Sprintf(" ansible_ssh_user=%s ansible_ssh_common_args=\"-o ConnectionAttempts=20\"", sshCredentials.user))
 		// Set with priority private key against password
-		if sshCredentials.privateKey != "" {
-			// TODO if not a path store it somewhere
-			// Note whould be better if we can use it directly https://github.com/ansible/ansible/issues/22382
+		if !e.cfg.UseSSHAgent && sshCredentials.privateKey != "" {
+			// check privateKey's a valid path
+			if is, err := pathutil.IsValidPath(sshCredentials.privateKey); err != nil || !is {
+				return errors.Errorf("%q is not a valid path", stringutil.Truncate(sshCredentials.privateKey, 20))
```

Comment:
Why truncate where?
We want to will typically want to get the invalid path and make `ls /my/deep/deep/path/to/key.pem ; ls /my/deep/deep/path/to; ls /my/deep/deep/path/; ls /my/deep/deep/` and so on to find the missing part.